src/environmentd/src/http.rs

```diff
@@ -322,203 +490,26 @@ pub async fn handle_leader_skip_catchup(
-            .route(
-                "/api/catalog/dump",
-                routing::get(catalog::handle_catalog_dump),
-            )
-            .route(
-                "/api/catalog/check",
-                routing::get(catalog::handle_catalog_check),
-            )
-            .route(
-                "/api/coordinator/check",
-                routing::get(catalog::handle_coordinator_check),
-            )
-            .route(
-                "/api/coordinator/dump",
-                routing::get(catalog::handle_coordinator_dump),
-            )
-            .route(
-                "/internal-console",
-                routing::get(|| async { Redirect::temporary("/internal-console/") }),
-            )
-            .route(
-                "/internal-console/*path",
-                routing::get(console::handle_internal_console),
-            )
-            .route(
-                "/internal-console/",
-                routing::get(console::handle_internal_console),
-            )
-            .layer(middleware::from_fn(internal_http_auth))
-            .layer(Extension(adapter_client_rx.clone()))
-            .layer(Extension(console_config))
-            .layer(Extension(active_connection_counter.clone()));
-
-        let ws_router = Router::new()
-            .route("/api/experimental/sql", routing::get(sql::handle_sql_ws))
-            // This middleware extracts the MZ user from the x-materialize-user http header.
-            // Normally, browser-initiated websocket requests do not support headers, however for the
-            // Internal HTTP Server the browser would be connecting through teleport, which should
-            // attach the x-materialize-user header to all requests it proxies to this api.
-            .layer(middleware::from_fn(internal_http_auth))
-            .with_state(WsState {
-                frontegg: Arc::new(None),
-                adapter_client_rx,
-                active_connection_counter,
-                helm_chart_version: helm_chart_version.clone(),
-            });
-
-        let leader_router = Router::new()
-            .route("/api/leader/status", routing::get(handle_leader_status))
-            .route("/api/leader/promote", routing::post(handle_leader_promote))
-            .route(
-                "/api/leader/skip-catchup",
-                routing::post(handle_leader_skip_catchup),
-            )
-            .with_state(deployment_state_handle);
-
-        let router = router
-            .merge(ws_router)
-            .merge(leader_router)
-            .apply_default_layers("internal", metrics);
-
-        InternalHttpServer { router }
+async fn x_materialize_user_header_auth(mut req: Request, next: Next) -> impl IntoResponse {
+    // TODO integrate this into the single auth flow
+    // or migrate teleport to basic auth.
+    if let Some(username) = req.headers().get("x-materialize-user").map(|h| h.to_str()) {
```

Comment:
Not needed in this PR but in the follow up if we are hanging onto x-materialize-user lets throw it in a constant